### PR TITLE
[new release] async-uri (0.4.0)

### PR DESCRIPTION
**Pull request by vbmithr**
Open Async (TLS) TCP connections with Uri.t

- Project page: <a href="https://github.com/vbmithr/async-uri">https://github.com/vbmithr/async-uri</a>
- Documentation: <a href="https://vbmithr.github.io/async-uri">https://vbmithr.github.io/async-uri</a>

##### CHANGES:

- Update to async v0.16

### CODE REVIEW

File: packages/async-uri/async-uri.0.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+license: "ISC"
+tags: ["async" "uri" "tls" "ssl"]
+homepage: "https://github.com/vbmithr/async-uri"
+doc: "https://vbmithr.github.io/async-uri"
+bug-reports: "https://github.com/vbmithr/async-uri/issues"
+depends: [
+  "ocaml"
+  "dune" {>= "3.10"}
+  "uri-sexp"
+  "core" {>= "0.16"}
```
Comment (haochenx) [inline packages/async-uri/async-uri.0.4.0/opam:18]:
```suggestion
  "core" {>= "v0.16.0"}
```

File: packages/async-uri/async-uri.0.4.0/opam [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "async" {>= "0.16"}
```
Comment (haochenx) [inline packages/async-uri/async-uri.0.4.0/opam:19]:
```suggestion
  "async" {>= "v0.16.0"}
```

File: packages/async-uri/async-uri.0.4.0/opam [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  "async_ssl" {>= "0.16"}
```
Comment (haochenx) [inline packages/async-uri/async-uri.0.4.0/opam:20]:
```suggestion
  "async_ssl" {>= "v0.16.0"}
```

**Review by haochenx — COMMENTED** [review]
```
[ERROR] The compilation of async-uri.0.4.0 failed at "dune build -p async-uri -j 31 @install".

#=== ERROR while compiling async-uri.0.4.0 ====================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/vbmithr/async-uri/releases/download/0.4.0/async-uri-0.4.0.tbz)
# path                 ~/.opam/4.04/.opam-switch/build/async-uri.0.4.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p async-uri -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/async-uri-7-322df8.env
# output-file          ~/.opam/log/async-uri-7-322df8.out
### output ###
# File "/home/opam/.opam/4.04/lib/ppx_inline_test/ppx_inline_test.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.04/lib/ppx_expect/ppx_expect.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.async_uri.objs/byte -I /home/opam/.opam/4.04/lib/async -I /home/opam/.opam/4.04/lib/async_extra -I /home/opam/.opam/4.04/lib/async_kernel -I /home/opam/.opam/4.04/lib/async_rpc_kernel -I /home/opam/.opam/4.04/lib/async_ssl -I /home/opam/.opam/4.04/lib/async_ssl/bindings -I /home/opam/.opam/4.04/lib/async_unix -I /home/opam/.opam/4.04/lib/base -I /home/opam/.opam/4.04/lib/base/caml -I /home/opam/.opam/4.04/lib/base/md5 -I /home/opam/.opam/4.04/lib/base/shadow_stdlib -I /home/opam/.opam/4.04/lib/bin_prot -I /home/opam/.opam/4.04/lib/bin_prot/shape -I /home/opam/.opam/4.04/lib/core -I /home/opam/.opam/4.04/lib/core_kernel -I /home/opam/.opam/4.04/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.04/lib/ctypes -I /home/opam/.opam/4.04/lib/fieldslib -I /home/opam/.opam/4.04/lib/integers -I /home/opam/.opam/4.04/lib/jane-street-headers -I /home/opam/.opam/4.04/lib/ocaml/threads -I /home/opam/.opam/4.04/lib/parsexp -I /home/opam/.opam/4.04/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_expect/collector -I /home/opam/.opam/4.04/lib/ppx_expect/common -I /home/opam/.opam/4.04/lib/ppx_expect/config -I /home/opam/.opam/4.04/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_inline_test/config -I /home/opam/.opam/4.04/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.04/lib/protocol_version_header -I /home/opam/.opam/4.04/lib/re -I /home/opam/.opam/4.04/lib/re/posix -I /home/opam/.opam/4.04/lib/seq -I /home/opam/.opam/4.04/lib/sexplib -I /home/opam/.opam/4.04/lib/sexplib/unix -I /home/opam/.opam/4.04/lib/sexplib0 -I /home/opam/.opam/4.04/lib/spawn -I /home/opam/.opam/4.04/lib/splittable_random -I /home/opam/.opam/4.04/lib/stdio -I /home/opam/.opam/4.04/lib/stdlib-shims -I /home/opam/.opam/4.04/lib/stringext -I /home/opam/.opam/4.04/lib/typerep -I /home/opam/.opam/4.04/lib/uri -I /home/opam/.opam/4.04/lib/uri-sexp -I /home/opam/.opam/4.04/lib/uri/services -I /home/opam/.opam/4.04/lib/variantslib -no-alias-deps -o src/.async_uri.objs/byte/async_uri.cmi -c -intf src/async_uri.mli)
# File "src/async_uri.mli", line 17, characters 36-53:
# Error: Unbound module Time_float
```

The lower bounds for janestreet packages seem incorrect.

```

<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build async-uri 0.4.0
+- 
+- The following changes have been performed
| - install async                   v0.11.0
| - install async_extra             v0.11.1
| - install async_kernel            v0.11.1
| - install async_rpc_kernel        v0.11.0
| - install async_ssl               v0.11.0
| - install async_unix              v0.11.0
| - install base                    v0.11.1
| - install base-num                base
...
| - install core                    v0.11.3
| - install core_kernel             v0.11.1
...
```

**Review by haochenx — APPROVED** [review]
Thanks! It now looks all good to me.

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Pull request merged**